src/main/java/com/example/AllPermutationPrime.java

```diff
@@ -0,0 +1,62 @@
+package com.example;
+
+import java.util.ArrayList;
+
+/**
+ * Készíts egy olyan függvényt, aminek
+ * bemenő paramétere n,
+ * és válaszként kilistázza azokat a számokat 1-től n-ig,
+ * amelyeknek az összes körkörös permutációja prím.
+ */
+public class AllPermutationPrime {
+
+    private static boolean isPrime(int n) {        
+        boolean prime = true;
+        int divider = n - 1;
+        while ((divider > 1) && prime) {
+            if (n % divider == 0) {
+                prime = false;
+            }
+            else {
+                divider--;
+            }
+        }
+        return prime;
+    }
+
+    private static int[] generateCircledPermutations(int checkedNumber) {
+        String number = String.valueOf(checkedNumber);       
+        int numberOfDigits = number.length();
+        number += number;
+        int[] permutations = new int[numberOfDigits];
+        for (int i = 0; i < numberOfDigits; i++) {
+            permutations[i] = Integer.parseInt(number.substring(i, i + numberOfDigits));
+        }
+        return permutations;
+    }
+
+    public static int[] allCircledPermutationsArePrimes(int n) {
+        ArrayList<Integer> goodNumbers = new ArrayList<>();
+        for (int checkedNumber = 2; checkedNumber <= n; checkedNumber++) {
```

Comment:
This code looks good to me but for the sake of exercise, can we rewrite it with streams?

It would look like something this:

```
<do something to generate a stream of integers 2 ... n>
.map(i -> generateCircledPermutations(i))
.filter(ints -> <you need a function here which checks if all permutations prime> )
.map(ints -> ints.get(0)) // pick just one number from the permutations
. <do something to collect all the numbers into a list>
```

Reply:
Done.